server/src/app.ts

```diff
@@ -19,12 +20,16 @@ app.use((req, res) => res.sendFile(path.join(__dirname, '..', '..', 'client', 'b
 // Express configuration
 app.set('port', process.env.PORT || 3055)
 
-Blockchain.updateData()
-Blockchain.ready.then(() => {
-  setInterval(() => {
-    console.log('********Start update data**************')
-    Blockchain.updateData()
-  }, conf.timeUpdate)
-})
+const job = new CronJob(conf.cronTimeExpression, function() {
```

Comment:
1. destructurize conf above:

```
const { cronTimeExpression, cronTimeZone } = conf

const job = new CronJob(cronTimeExpression, ..., cronTImeZone, ...)
```

2. crontask function could be arrow as don't access the task object via `this.` (and if you'll need this you could using `const job` declared

3. pass arrow function `() => job.stop()` to the exit eventz handler (or you'll lose context)
also you could use one-liner e.g. `['SIGINT', 'SIGTERM', 'exit'].forEach(...`

Reply:
5.  you haven't added lock preventing task to be executed more that once at the time. It's low probability but better to use it anyway:
```javascript
import AsyncLock from 'async-lock'

const lock = new AsyncLock()


const job = new CronJob(cronTimeExpression, async () => {
  try {
    await lock.acquire('updateData', async () => {
      console.log('******** Start update data **************')
      await Blockchain.updateData()
    })
    // log success
  } catch (exception) {
    // log error
  } finally {
    // log update finished
  } 
}, ... )
```

also it would be good to add try catch for logging